cluster-autoscaler/main.go

```diff
@@ -278,7 +278,10 @@ func buildAutoscaler() (core.Autoscaler, error) {
 	autoscalingOptions := createAutoscalingOptions()
 	kubeClient := createKubeClient(getKubeConfig())
 	eventsKubeClient := createKubeClient(getKubeConfig())
+
 	processors := ca_processors.DefaultProcessors()
+	processors.PodListProcessor = core.NewFilterOutSchedulablePodListProcessor()
```

Comment:
This should really be part of DefaultProcessors. I guess it's tricky due to cyclic dependencies though, right? If so it's not worth refactoring everything just to put it there.

Reply:
I will send ahother PR with just moving things around (and no logic changes) when we decide this one is good to go